code/datums/components/regenerator.dm

```diff
@@ -129,6 +137,11 @@
 /datum/component/regenerator/proc/should_be_regenning(mob/living/who)
 	if(who.stat == DEAD)
 		return FALSE
+
+	if(regen_check && !regen_check.Invoke(who))
```

Comment:
I don't love that this proc is developing side-effects but I guess there's not a better place to put it.